lib/ASRouterTargeting.jsm

```diff
@@ -168,6 +168,9 @@ const TargetingGetters = {
       Cu.reportError("Problem parsing JSON message provider pref for ASRouter");
     }
     return cohorts;
+  },
+  get totalBookmarksCount() {
+    return asProvider.getTotalBookmarksCount();
```

Comment:
Would you mind adding some caching to this similar to how `TopFrecentSitesCache` works?